### PR TITLE
refactor: replace filename hyphens with underscores

### PR DESCRIPTION
**Pull request by innovaker**
Aligns *.h and *.c to underscore naming convention.

Note: this would be a minor breaking change for users with custom boards or shields.

### CODE REVIEW

**Comment by innovaker** [comment]
I hit this whilst refactoring a related part of the system.  I wasn't sure if there was a rationale for having two conventions implementation.

**Review by petejohanson — CHANGES_REQUESTED** [review]
This is a good call, but to avoid breaking external boards, we should preserve the old filename, but with a single line content that imports the proper named file, along with a warning/deprecation notice. Keep local build clean, help keep external boards/shields from breaking.

**Comment by innovaker** [comment]
> This is a good call, but to avoid breaking external boards, we should preserve the old filename, but with a single line content that imports the proper named file, along with a warning/deprecation notice. Keep local build clean, help keep external boards/shields from breaking.

Good idea.  I've added these files:
- `kscan-mock.h`
- `matrix-transform.h`

And tested them by stripping out the commits for the boards, shields, and tests here:
- https://github.com/innovaker/zmk/actions/runs/437888743
- https://github.com/innovaker/zmk/actions/runs/437888742

File: app/include/dt-bindings/zmk/kscan-mock.h [hunk 1]
```diff
@@ -4,11 +4,4 @@
...
 
-#pragma once
-
-#define ZMK_MOCK_IS_PRESS(v) ((v & (0x01 << 31)) != 0)
-#define ZMK_MOCK_PRESS(row, col, msec) (row + (col << 8) + (msec << 16) + (0x01 << 31))
-#define ZMK_MOCK_RELEASE(row, col, msec) (row + (col << 8) + (msec << 16))
-#define ZMK_MOCK_ROW(v) (v & 0xFF)
-#define ZMK_MOCK_COL(v) ((v >> 8) & 0xFF)
-#define ZMK_MOCK_MSEC(v) ((v & ~(0x01 << 31)) >> 16)
+#include "kscan_mock.h"
```
Comment (petejohanson) [inline app/include/dt-bindings/zmk/kscan-mock.h:7]: Can we add a warning above this: https://stackoverflow.com/a/3826868

File: app/include/dt-bindings/zmk/matrix-transform.h [hunk 2]
```diff
@@ -4,7 +4,4 @@
  * SPDX-License-Identifier: MIT
  */
 
-#define KT_ROW(item) (item >> 8)
-#define KT_COL(item) (item & 0xFF)
-
-#define RC(row, col) (((row) << 8) + (col))
\ No newline at end of file
+#include "matrix_transform.h"
```
Comment (petejohanson) [inline app/include/dt-bindings/zmk/matrix-transform.h:7]: Ditto

**Review by petejohanson — CHANGES_REQUESTED** [review]

File: app/include/dt-bindings/zmk/kscan-mock.h (hunk 1 above)
Comment (innovaker) [reply app/include/dt-bindings/zmk/kscan-mock.h:7]: Something like this?
`#warning "'kscan-mock.h' has been deprecated and replaced by 'kscan_mock.h'"`

File: app/include/dt-bindings/zmk/matrix-transform.h (hunk 2 above)
Comment (innovaker) [reply app/include/dt-bindings/zmk/matrix-transform.h:7]: `#warning "'matrix-transform.h' has been deprecated and replaced by 'matrix_transform.h'"`

File: app/include/dt-bindings/zmk/kscan-mock.h (hunk 1 above)
Comment (innovaker) [reply app/include/dt-bindings/zmk/kscan-mock.h:7]: >  along with a warning/deprecation notice
apologies for missing that btw

Comment (innovaker) [reply app/include/dt-bindings/zmk/kscan-mock.h:9]: Done, tested here:
- https://github.com/innovaker/zmk/actions/runs/438462011
- https://github.com/innovaker/zmk/actions/runs/438462009

File: app/include/dt-bindings/zmk/matrix-transform.h (hunk 2 above)
Comment (innovaker) [reply app/include/dt-bindings/zmk/matrix-transform.h:9]: Ditto above.

**Review by petejohanson — APPROVED** [review]
Thanks!

**Pull request merged**